### PR TITLE
feat(seo): add sitemap.xml and robots.txt

### PR DESCRIPTION
**Pull request by simon-archer**
# Changes
- Added sitemap.xml and robots.txt generation using next-sitemap
- Configured postbuild script for automatic generation after builds
- Added next-sitemap configuration file
- Tested locally and verified outputs

# Testing
- [x] Ran `yarn build:storefront` and verified sitemap.xml and robots.txt are generated
- [x] Verified sitemap.xml contains correct URLs
- [x] Verified robots.txt contains correct configuration
- [x] Tested accessibility at localhost:3000/robots.txt and localhost:3000/sitemap.xml

### CODE REVIEW

File: apps/storefront/public/robots.txt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Disallow: /index.php/comment/reply/
+Disallow: /index.php/filter/tips
+Disallow: /index.php/node/add/
+Disallow: /index.php/search/
+Disallow: /index.php/user/password
+Disallow: /index.php/user/register
+Disallow: /index.php/user/login
+Disallow: /index.php/user/logout
+Disallow: /index.php/media/oembed
+Disallow: /index.php/*/media/oembed
```
Comment (Barsnes) [inline apps/storefront/public/robots.txt:73]:
```suggestion
```
This is specifically for Drupal, so we don't need this in our file

Comment (Barsnes) [reply apps/storefront/public/robots.txt:73]: I think we could do
`Allow: *` 🤔

Comment (simon-archer) [reply apps/storefront/public/robots.txt:73]: Thanks for the code suggestions. I don't have enough knowledge about the implications of `Allow: * ` but happy to make the changes if it is agreed upon by the others.

File: apps/storefront/public/robots.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#
+# This file will be ignored unless it is at the root of your host:
+# Used:    http://example.com/robots.txt
+# Ignored: http://example.com/site/robots.txt
+#
+# For more information about the robots.txt standard, see:
+# http://www.robotstxt.org/robotstxt.html
+
+User-agent: *
+# CSS, JS, Images
```
Comment (Barsnes) [inline apps/storefront/public/robots.txt:17]:
```suggestion
Allow: /
```

**Pull request merged**